convert elisp packages to melpaBuild

AndersonTorres opened the pull request:
## Description of changes

Following https://github.com/NixOS/nixpkgs/issues/278925

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres committed:
emacs.pkgs.sunrise-commander: convert to melpaBuild

AndersonTorres committed:
emacs.pkgs.yes-no: convert to melpaBuild

AndersonTorres committed:
emacs.pkgs.git-undo: convert to melpaBuild

AndersonTorres committed:
emacs.pkgs.evil-markdown: convert to melpaBuild

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/sunrise-commander/default.nix:
```diff
@@ -1,24 +1,43 @@
   pname = "sunrise-commander";
-  version = "unstable=2021-09-27";
+  ename = "sunrise";
+  version = "20210924.1620";
```
`24` is probably wrong because it should be within `[27-1, 27+1]`.

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/yes-no/default.nix:
```diff
@@ -1,19 +1,36 @@
+  src = fetchFromGitHub {
+    owner = "emacsmirror";
+    repo = "emacswiki.org";
+    rev = "02dbe9aa48c735cfcc7e977ea8fe512328666cad";
```
Using the old commit seems more reasonable to me because `yes-no.el` has not changed since then.

AndersonTorres replied:
I am using the most recent commit here, since it is easier (no need for pushing that specific commit) and goes in line with the idea of "track the most recent commit".

Further, it can be more useful if anyone is interested on other single-file packages from ewiki.

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/sunrise-commander/default.nix:
```diff
@@ -1,24 +1,43 @@
   ];
 
+  recipe = writeText "recipe" ''
+    (sunrise
```
This needs to be the same as `ename` (`ename` defaults to `pname`).

Is this better?

```suggestion
    (${ename}
```

AndersonTorres replied:
I have a better idea: a `mkMelpaRecipe` function!

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/evil-markdown/default.nix:
```diff
@@ -1,34 +1,42 @@
     hash = "sha256-HBBuZ1VWIn6kwK5CtGIvHM1+9eiNiKPH0GUsyvpUVN8=";
   };
+in
+melpaBuild rec {
```
I guess one purpose of your `let ... in` is to avoid `rec` and this one is indeed redundant.

```suggestion
melpaBuild {
```

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/sunrise-commander/default.nix:
```diff
@@ -1,24 +1,43 @@
-  buildInputs = [
+  outputs = [ "out" "doc" ];
+
+  packageRequires = [
```
This probably does not needed.

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/evil-markdown/default.nix:
```diff
@@ -1,34 +1,42 @@
   pname = "evil-markdown";
-  version = "0.pre+unstable=2021-07-21";
-
+  ename = "evil-markdown";
```
`ename` defaults to `pname` so this is redundant.

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/manual-packages/git-undo/default.nix:
```diff
@@ -1,25 +1,41 @@
+melpaBuild {
+  inherit pname ename version src;
+
+  packageRequires = [
+    emacs
+  ];
```
probably not needed

AndersonTorres committed:
emacs.pkgs.consult-gh: update to new format

jian-lin commented on pkgs/applications/editors/emacs/elisp-packages/updater-emacs.nix:
```diff
@@ -1,33 +1,8 @@
-  in [ promise semaphore ]);
-
+  emacsEnv = pkgs.emacs.pkgs.withPackages(epkgs: with epkgs; [
+    promise semaphore ]);
```
Does the manual update script still work after this change?  I made this change a few months ago and found that this change broke the manual update script.

I am not sure if this change is desirable because pinning some dependencies makes the manual update script more reproducible.

jian-lin replied:
Oh, just find that you add the pinned version of `async-await`, `promise` and `semaphore` to manual packages, which shadows newer version of them on MELPA.  I do not think that is a good idea.

AndersonTorres replied:
They are on melpa??
I thougth they weren't...

Why pinning them? Those packages are so stable they have no changes for years...

Nonetheless it looks cleaner to melpaize them...

jian-lin replied:
> Why pinning them?

There were a few patches not in the upstream promise packages back then.

AndersonTorres replied:
For now I will merely unpin them.

AndersonTorres committed:
[wip] emacs.pkgs.consult-gh: use writeMelpaRecipe

AndersonTorres committed:
[wip] emacs.pkgs.writeMelpaRecipe: init

A helper for writing MELPA-compatible recipes.
The argument names are exactly the same as the ones from MELPA documentation:
https://github.com/melpa/melpa#recipe-format

AndersonTorres committed:
[wip] emacs.pkgs.sunrise-commander: use writeMelpaRecipe

AndersonTorres committed:
[wip] emacs.pkgs.yes-no: use writeMelpaRecipe

AndersonTorres committed:
[wip] emacs.pkgs.git-undo: use writeMelpaRecipe

AndersonTorres committed:
[wip] emacs.pkgs.evil-markdown: use writeMelpaRecipe

AndersonTorres commented:
Too much conflicts. Let's restart.
